amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/HierarchicalShardSyncerTest.java

```diff
@@ -1229,92 +1390,198 @@ public void testDetermineNewLeasesToCreateSplitMergeLatest1() {
-        final Set<ExtendedSequenceNumber> expectedCheckpoint = new HashSet<>(
-                Collections.singletonList(ExtendedSequenceNumber.TRIM_HORIZON));
+    /*
+     * Shard structure (x-axis is epochs):
+     * 0  3   6   9
+     * \ / \ / \ /
+     *  2   5   8
+     * / \ / \ / \
+     * 1  4   7  10
+     *
+     * Current leases: (5)
+     */
+    @Test
+    public void testDetermineNewLeasesToCreateSplitMergeLatestB_CompleteHashRange() {
+        final List<Shard> shards = constructShardListForGraphB();
+        final List<String> shardIdsOfCurrentLeases = Arrays.asList("shardId-5");
+        final Map<String, ExtendedSequenceNumber> expectedShardIdCheckpointMap = Collections.emptyMap();
+        assertExpectedLeasesAreCreated(shards, shardIdsOfCurrentLeases, INITIAL_POSITION_LATEST, expectedShardIdCheckpointMap);
+    }
 
-        assertThat(newLeases.size(), equalTo(expectedLeaseKeys.size()));
-        assertThat(checkpoints.size(), equalTo(expectedLeaseKeys.size()));
-        assertThat(leaseKeys, equalTo(expectedLeaseKeys));
-        assertThat(checkpoint, equalTo(expectedCheckpoint));
+    /*
+     * Shard structure (x-axis is epochs):
+     * 0  3   6   9
+     * \ / \ / \ /
+     *  2   5   8
+     * / \ / \ / \
+     * 1  4   7  10
+     *
+     * Current leases: (0, 1, 2, 3, 4, 5)
+     */
+    @Test
+    public void testDetermineNewLeasesToCreateSplitMergeLatestB_CompleteHashRangeWithoutGC() {
+        final List<Shard> shards = constructShardListForGraphB();
+        final List<String> shardIdsOfCurrentLeases = Arrays.asList("shardId-0", "shardId-1", "shardId-2", "shardId-3",
+                "shardId-4", "shardId-5");
+        final Map<String, ExtendedSequenceNumber> expectedShardIdCheckpointMap = Collections.emptyMap();
+        assertExpectedLeasesAreCreated(shards, shardIdsOfCurrentLeases, INITIAL_POSITION_LATEST, expectedShardIdCheckpointMap);
     }
 
-//    /**
-//     * Test CheckIfDescendantAndAddNewLeasesForAncestors (initial position TrimHorizon)
-//     * Shard structure (each level depicts a stream segment):
-//     * 0 1 2 3 4   5- shards till epoch 102
-//     * \ / \ / |   |
-//     *  6   7  4   5- shards from epoch 103 - 205
-//     *   \ /   |  /\
-//     *    8    4 9  10 - shards from epoch 206 (open - no ending sequenceNumber)
-//     * Current leases: (4, 5, 7)
-//     */
+    /*
+     * Shard structure (x-axis is epochs):
+     * 0  3   6   9
+     * \ / \ / \ /
+     *  2   5   8
+     * / \ / \ / \
+     * 1  4   7  10
+     *
+     * Current leases: empty set
+     */
     @Test
-    public void testDetermineNewLeasesToCreateSplitMergeHorizon2() {
-        final List<Shard> shards = constructShardListForGraphA();
-        final List<Lease> currentLeases = Arrays.asList(newLease("shardId-4"), newLease("shardId-5"),
-                newLease("shardId-7"));
+    public void testDetermineNewLeasesToCreateSplitMergeLatestB_EmptyLeaseTable() {
```

Comment:
can we have trim, at_ts equivalent of this?

Reply:
Yep we have these below on lines 1814 and 2033